src/util-dpdk-net_bonding.c

```diff
@@ -0,0 +1,151 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Lukas Sismis <lukas.sismis@gmail.com>
+ */
+
+#ifndef UTIL_DPDK_NET_BONDING_C
+#define UTIL_DPDK_NET_BONDING_C
+
+#include "util-dpdk-net_bonding.h"
+#include "suricata-common.h"
+
+#ifdef HAVE_DPDK
+
+#include "util-dpdk.h"
+#include "util-debug.h"
+
+/**
+ * Determines if the port is Bond or not by evaluating device driver name
+ * @param pid port ID
+ * @return 0 - the device si Bond PMD, 1 - regular device, <0 error
+ */
+int32_t net_bonding_is_port_bond(uint16_t pid)
+{
+    int32_t ret;
+    struct rte_eth_dev_info di;
+    ret = rte_eth_dev_info_get(pid, &di);
+    if (ret < 0) {
+        char dev_name[RTE_ETH_NAME_MAX_LEN];
+        int32_t subret = rte_eth_dev_get_name_by_port(pid, dev_name);
+        if (ret < 0) {
+            SCLogError("Port %d: unable to get port name (err: %s)", pid, rte_strerror(-ret));
+            return subret;
+        }
+        SCLogError("%s: unable to get device info (err: %s)", dev_name, rte_strerror(-ret));
+        return ret;
+    }
+
+    return strcmp(di.driver_name, "net_bonding") == 0 ? 0 : 1;
+}
+
+uint16_t net_bonding_get_bonded_devices(
+        uint16_t bond_pid, uint16_t bonded_devs[], uint16_t bonded_devs_length)
+{
+    int32_t len;
+    len = rte_eth_bond_slaves_get(bond_pid, bonded_devs, bonded_devs_length);
```

Comment:
declare here

Reply:
please review rest of the PR for this pattern

Reply:
done